### PR TITLE
Update NonEmptyAtclauseDescription documentation (Issue #7604)

### PR DESCRIPTION
**Pull request by coderdeadpool**
Issue:#7604
Added example for each existing config to show what code is ok and what code cause violation.
``` mvn clean verify ``` passed.

Web Page:
Screenshot (25)
Screenshot (26)
Screenshot (17)
Screenshot (18)
Screenshot (19)
Screenshot (20)
Screenshot (21)

CLI:
First Example:
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="PARAM_LITERAL,DEPRECATED_LITERAL,THROWS_LITERAL"/>
    </module>
  </module>
</module>
```
```
$ cat test.java
class Test
{
  /**
  * Violation for param "b" and at tags "deprecated", "throws".
  * @param a Some javadoc // OK
  * @param b
  * @deprecated
  * @throws Exception
  */
  public int method(String a, int b) throws Exception
  {
    return 1;
  }
}

```
```
$ java -jar target/checkstyle-10.8.2-SNAPSHOT-all.jar -c config.xml test.java
Starting audit...
[ERROR] C:\Users\hp\checkstyle\checkstyle\test.java:6: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] C:\Users\hp\checkstyle\checkstyle\test.java:7: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
[ERROR] C:\Users\hp\checkstyle\checkstyle\test.java:8: At-clause should have a non-empty description. [NonEmptyAtclauseDescription]
Audit done.
Checkstyle ends with 3 errors.
```
Second Example:
``` 
$ cat config1.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="PARAM_LITERAL"/>
    </module>
  </module>
</module>
```
``` 
$ cat test1.java
class Test
{
  /**
  * Violation for param "b".
  * <code>@param</code> a Some javadoc // OK
  * <code>@param</code> b

  */
  public int method(String a, int b)
  {
    return 1;
  }
}
```
``` 
$ java -jar target/checkstyle-10.8.2-SNAPSHOT-all.jar -c config1.xml test1.java
Starting audit...
Audit done.
```
Third Example:
```
$ cat config2.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="RETURN_LITERAL"/>
    </module>
  </module>
</module>
```
```
$ cat test2.java
class Test
{
  /**
  * Violation for return tag.
  * <code>@param</code> a Some javadoc
  * <code>@return</code>

  */
  public int method(String a)
  {
    return 1;
  }
}
```
```
$ java -jar target/checkstyle-10.8.2-SNAPSHOT-all.jar -c config2.xml test2.java
Starting audit...
Audit done.
```
Fourth Example:
```
$ cat config3.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="THROWS_LITERAL"/>
    </module>
  </module>
</module>
```
```
$ cat test3.java
class Test
              {
              /**
              * Violation for throws tag.
              * <code>@param</code> a Some javadoc
              * <code>@throws</code> Exception

              */
              public int method(String a) throws Exception
              {
              return 1;
              }
              }
```
```
$ java -jar target/checkstyle-10.8.2-SNAPSHOT-all.jar -c config3.xml test3.java
Starting audit...
Audit done.
```
Fifth Example:
```
$ cat config4.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="DEPRECATED_LITERAL"/>
    </module>
  </module>
</module>
```
```
$ cat test4.java
class Test
              {
              /**
              * Violation for deprecated tag.
              * <code>@param</code> a Some javadoc
              * <code>@deprecated</code>

              */
              public int method(String a)
              {
              return 1;
              }
              }
```
```
$ java -jar target/checkstyle-10.8.2-SNAPSHOT-all.jar -c config4.xml test4.java
Starting audit...
Audit done.
```
Sixth Example:
```
$ cat config5.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="NonEmptyAtclauseDescription">
        <property name="javadocTokens" value="EXCEPTION_LITERAL"/>
    </module>
  </module>
</module>
```
```
$ cat test5.java
class Test
              {
              /**
              * Violation for exception tag.
              * <code>@param</code> a Some javadoc
              * <code>@exception</code> Exception

              */
              public int method(String a) throws Exception
              {
              return 1;
              }
              }
```
```
$ java -jar target/checkstyle-10.8.2-SNAPSHOT-all.jar -c config5.xml test5.java
Starting audit...
Audit done.
```

### CODE REVIEW

**Comment by romani** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
> fix CI failures, commit message should be in a strict format.
I am on it

**Comment by coderdeadpool** [comment]
> fix CI failures, commit message should be in a strict format.

I am on it

**Comment by coderdeadpool** [comment]
@romani CI failures fixed

**Comment by coderdeadpool** [comment]
@romani Do i have to add web page screenshot and CLI to pass all the checks sucessfully ?

**Comment by romani** [comment]
@coderdeadpool , share CLI outputs for all example to prove that it works as you write in example.

example: https://github.com/checkstyle/checkstyle/pull/12805

**Comment by coderdeadpool** [comment]
And the image file too ?

**Comment by romani** [comment]
yes, as auto website generation is not working, not sure why.

**Comment by coderdeadpool** [comment]
As far as i understand, its may be because the spring boot plugins are not present in the pom.xml file

**Comment by romani** [comment]
do
https://github.com/checkstyle/checkstyle/pull/12834#issuecomment-1467948409

**Comment by coderdeadpool** [comment]
Github, generate report

**Comment by romani** [comment]
your triggering is skipped - https://github.com/checkstyle/checkstyle/actions/workflows/site.yml
do it manually.

**Comment by coderdeadpool** [comment]
@romani  I am on it.

**Comment by coderdeadpool** [comment]
@romani I have generated the website manually. What to do next ?

**Comment by coderdeadpool** [comment]
@romani how do i generate CLI output or where do i find it ?

**Comment by romani** [comment]
You manually create Test.java and run it, see example that I gave you.

**Comment by coderdeadpool** [comment]
@romani added the image of the webpage in PR description.

**Comment by coderdeadpool** [comment]
@romani Added CLI outputs in the PR description.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java [hunk 1]
```diff
@@ -55,11 +55,12 @@
  * </li>
  * </ul>
  * <p>
- * To configure the default check that will check {@code @param}, {@code @return},
- * {@code @throws}, {@code @deprecated}:
+ * To configure the check to validate {@code @param} tags:
  * </p>
  * <pre>
- * &lt;module name="NonEmptyAtclauseDescription"/&gt;
+ * &lt;module name="NonEmptyAtclauseDescription"&gt;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:61]: Keep default config and example of execution in addition to what you suggest.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 1 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:61]: @romani  Like this:
```
*<p>
* To configure the default check that will check {@code @param}, {@code @return},
 * {@code @throws}, {@code @deprecated}:
 * </p>
 * <pre>
 * &lt;module name="NonEmptyAtclauseDescription"/&gt;
 * </pre>
 * <p>
 * Example:
 * </p>
 * <pre>
 * class Test
 * {
 * &#47;**
 * * Violation for param "b" and at tags "deprecated", "throws".
 * * &#64;param a Some javadoc // OK
 * * &#64;param b
 * * &#64;deprecated
 * * &#64;throws Exception
 * *&#47;
 * public int method(String a, int b) throws Exception
 * {
 * return 1;
 * }
 * }
 * </pre>
 ```

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:61]: Yes

Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:61]: I am on it.

Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:87]: @romani Added changes you asked for.

**Comment by coderdeadpool** [comment]
@romani Added Images of webpage and CLI outputs of the updated changes

**Comment by coderdeadpool** [comment]
#12834

**Comment by coderdeadpool** [comment]
Github, generate website

**Comment by coderdeadpool** [comment]
Github, generate report

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java [hunk 2]
```diff
@@ -95,13 +95,111 @@
...
+ *   &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class Test
+ * {
+ * &#47;**
+ * * Violation for throws tag.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:148]: lets make javadoc content to be same in all examples.
please use single line comments to mark violation

```
class Test
{
  // violation 4 lines below on 'return' tag.
  /**
  * Some summary.
  *  @param a Some javadoc
  * @return
```

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java [hunk 3]
```diff
@@ -95,13 +95,111 @@
...
  * &#47;**
- * * Violation for param "b". Tags "deprecated", "throws" are ignored.
+ * * Violation for param "b".
  * * &#64;param a Some javadoc // OK
  * * &#64;param b
- * * &#64;deprecated
+ * *&#47;
+ * public int method(String a, int b)
+ * {
+ * return 1;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:104]: please use proper indentation in javadoc to let it be read by humans easily even in javadoc format
https://github.com/checkstyle/checkstyle/blob/fe36c48a8b75bb97f5b4c617e0a79dfb724ead0a/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java#L136-L139

Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:104]: @romani  Like this
```
/**
  * Violation for param "b" and at tags "deprecated", "throws".
  * @param a Some javadoc // OK
  * @param b
  * @deprecated
  * @throws Exception
  */
public int method(String a, int b) throws Exception {
  return 1;
}
```

Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:104]: And do it like this to all examples i added ?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java [hunk 4]
```diff
@@ -95,13 +95,111 @@
...
+ * </p>
+ * <pre>
+ * class Test
+ * {
+ * &#47;**
+ * * Violation for return tag.
+ * * &#64;param a Some javadoc
+ * * &#64;return
+ * *&#47;
+ * public int method(String a)
```
[the remark this answers is not in the document]
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: @romani  You mean to use the same java code and javadoc comment like this
```
class Test
 * {
 * &#47;**
 * * Violation for param "b" and at tags "deprecated", "throws".
 * * &#64;param a Some javadoc // OK
 * * &#64;param b
 * * &#64;deprecated
 * * &#64;throws Exception
 * *&#47;
 * public int method(String a, int b) throws Exception
 * {
 * return 1;
 * }
 * }
```
in each individual examples.

[the remark this answers is not in the document]
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: And if i use this in each individual example, i don't think we need to make java code with two methods that is used in all examples. Please correct me if am wrong.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 2 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:148]: @romani I did'nt understand what are you suggesting. Are you saying just to use single line comment to mark voilation in each example.
like replace this  ``` * Violation for return tag. ```
with this ``` //violation ```

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:148]: Yes, attention to `4 lines belo`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: Ideally you should use one method, but you are limited to

[the remark this answers is not in the document]
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: So can i do it like the above code i showes. Will that be okay ?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 2 above)
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:148]: Ok, on it

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: Please do and let me see it again

[the remark this answers is not in the document]
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: @romani  After implementing what you asked, i got error message when i ran ``` mvn clean verify``` 
Screenshot (28)

Screenshot (29)

[the remark this answers is not in the document]
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: @romani I am not able to identify that what causing this error, Please help me out here.

[the remark this answers is not in the document]
Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: @coderdeadpool read carefully, looks you have misspelling in Violation. You can use `diff` command to help identify such issues

[the remark this answers is not in the document]
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: @nrmancuso "Violation" is spelled correctly. there is no such word called "Voilation". Correct me if am wrong

[the remark this answers is not in the document]
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:127]: @coderdeadpool , keep debugging it in your IDE to spot a problem

**Comment by romani** [comment]
Single commit always, and make CI green

**Comment by coderdeadpool** [comment]
@romani The changes you had asked for was done and ```mvn clean verify```  ran sucessfully, The build was sucesss. I added these commits by mistake, i wanted to commit the changes in the same commit, however this happen.
How to resolve this ? Should i create a new PR and close this one.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (coderdeadpool) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java]: @nrmancuso Yes you were right, it resolved when i changed "Violation" to "Voilation". But how ? can you explain please.

**Comment by coderdeadpool** [comment]
@romani  how should i delete the last two commits ?

**Comment by coderdeadpool** [comment]
@romani I was unable to squash all the commits into one. Should I create a new pull request for this issue?

**Comment by coderdeadpool** [comment]
@romani I did the changes you asked for and after running ```mvn clean verify`` command the build was sucess, But here some checks are not passing.

**Comment by romani** [comment]
@coderdeadpool , keep looking at CIs and fix problems that raise: spelling
https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=13595&view=logs&j=6a833393-c551-5000-0916-5000fd3b86e7&t=bdbbd129-77f0-5d73-ce7d-db58d34c8358&l=38

and rebase on latest out master code, here is how to do this https://checkstyle.org/beginning_development.html#Starting_Development

**Comment by romani** [comment]
Github, generate website

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java [hunk 5]
```diff
@@ -67,44 +67,153 @@
...
- * *&#47;
- * public int method(String a, int b) throws Exception
+ *  // Violation for param "b" and at tags "deprecated", "throws".
+ *  &#47;**
+ *  * Some summary.
+ *  * &#64;param a Some javadoc // OK
+ *  * &#64;param b
+ *  * &#64;deprecated
+ *  * &#64;throws Exception
+ *  *&#47;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java:77]: there is no `return` please add, you have it in others.
java doc comment should be idential by structure in all examples.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java]: change all to Violation, fix a typo.

**Comment by romani** [comment]
You have to rebase to latest our master.

Please share images of new docs in PR description.

**Comment by coderdeadpool** [comment]
@romani I on it.

**Comment by coderdeadpool** [comment]
@romani I have done all the steps mentioned in https://checkstyle.org/beginning_development.html#Starting_Development but still failed to rebase it, don't know why.

**Comment by coderdeadpool** [comment]
@romani closing this PR. And creating a new one with the changes.

**Pull request closed without merging**